kong/plugins/hmac-auth/access.lua

```diff
@@ -27,8 +29,71 @@ local DIGEST = "digest"
 local SIGNATURE_NOT_VALID = "HMAC signature cannot be verified"
 local SIGNATURE_NOT_SAME = "HMAC signature does not match"
 
+local new_tab
+do
+  local ok
+  ok, new_tab = pcall(require, "table.new")
+  if not ok then
+    new_tab = function() return {} end
+  end
+end
+
 local _M = {}
 
+local hmac = {
+  ["hmac-sha1"] = function(secret, data)
+    return crypto.hmac.digest("sha1", data, secret, true)
```

Comment:
Maybe we should keep using `ngx.sha1` when possible? This way we reduce the risk and we take advantage of using the FFI over the `crypto` lib?